### DISCUSSION TITLE
Slow updates/lag?

### DISCUSSION THREAD

**Discussion by davidcharlessmith516** [agronick/aa-torque#107] [Q&A] [answered]
AA Torque seems kind of slow to update values on the screen (speed limit, for instance), taking maybe a second or two between changes. I checked the realtime data in Torque Pro and it definitely updates more quickly than in AA Torque. Is there some setting I'm missing in either program for faster value updates?

**Comment by agronick** [comment]
Its polling every 200ms. You should be getting about 5 updates a second. I can add an option to lower the value. There is no way to get the data from Torque without polling though.

**Reply by davidcharlessmith516** [reply]
Based upon your reply, I'm going to try a different Bluetooth OBD adapter and see if that makes a difference. I don't think it's a BLE adapter, but I can't say for sure that it isn't since I can't find very many details about it anywhere. Thank you for the response. I'll let you know how it things go with the next adapter.

**Comment by davidcharlessmith516** [comment, marked as answer]
I tried a different Bluetooth OBD adapter and that seemed to improve the update speed. 

This first adapter I purchased from Amazon was a [$7.98 ELM 327 Bluetooth OBD2 adapter](https://www.amazon.com/dp/B0C8T76CPC). I could not find very many details about it but it's possible it was a BLE adapter. 

The second adapter that seems to be working better is a [$13.99 Veepak OBD2 Bluetooth adapter](https://a.co/d/78QXM52). 

The second adapter still doesn't seem to be updating the number on the screen five times per second (maybe two times per second), but this bit of troubleshooting tells me that I could be probably improve the update speed with an even better adapter.

**Comment by davidcharlessmith516** [comment]
This doesn't appear to be a programming issue, rather an adapter issue.
